### PR TITLE
Farts to have 3 second cooldown

### PR DESCRIPTION
**Pull request by TheTyrant25**
[CONTROVERSIAL] [INPUT]
## About the PR
Changes the fart cooldown from the default emote cooldown of one second to 3 seconds. Multiplies the fart gas quantities by approximately 3 to match.
Also, internally code wise, replace the time units in the `emote_check()` proc with proper time defines, e.g. `50` => `5 SECONDS` as well as using defines on the oxygen/plasma farts.
## Why's this needed?
Prevents spam of one of the more 'powerful' emotes, bringing it in line with things like flipping, which itself has 5 second cooldown. Makes each individual fart count more. Makes engineering using farts less spammy. Reduces noise spam.

## Changelog
```changelog
(u)Tyrant
(*)Farts now have a 3 second cooldown instead of 1 second. Fart volume is now 3 times more to make up for it.
```

### CODE REVIEW

File: code/mob/living/carbon/human/procs/emote.dm [hunk 1]
```diff
@@ -93,7 +93,7 @@
 
 			if ("fart")
 				var/oxyplasmafart = 0
-				if (src.emote_check(voluntary) && farting_allowed && (!src.reagents || !src.reagents.has_reagent("anti_fart")))
+				if (src.emote_check(voluntary, 5 SECONDS) && farting_allowed && (!src.reagents || !src.reagents.has_reagent("anti_fart")))
```
Comment (TheTyrant25) [inline code/mob/living/carbon/human/procs/emote.dm:96]: this is the salient part. all the other lines are just misc cleanup

**Comment by TheTyrant25** [comment]
does seem to be a commonly held view, yes

**Comment by TheTyrant25** [comment]
note that the chance of getting a foul smelling ad should also be buffed but that is in the secret submodule

**Comment by RSG250** [comment]
> note that the chance of getting a foul smelling ad should also be buffed but that is in the secret submodule
isnt the part that actually makes you fart it out NOT In the secret module, cause last i checked I would ocassionallyfart out a blank paper local, which was months ago sooo

also side note, as number 1 farter aka one whole presses the fart button whenever i can, a 5 second cooldown saddens me.

**Comment by RSG250** [comment]
oh and before i forget, this wont cause other emotes to go on cooldown right? cause if im remembering right, FLIPPING cause all other emotes to not be used until its cooldown is over ((again, last i checked at least) i might be confusing uhhhh when critters emote but... just to be safe). and personally i find that rather annoying

**Comment by hobartbartholomewpepperidgejr** [comment]
man i still dont understand why people want to remove farts

**Comment by pgmzeta** [comment]
How/Does this impact buttcrab balance?

**Comment by Sbmhawk** [comment]
This would heavily impact buttcrabs and high decay digestion.

**Comment by deathrobotpunch** [comment]
PLEASE GOD I BEG OF YOU DO NOT IMPLEMENT THAT NERF

**Comment by TheTyrant25** [comment]
it doesnt affect buttcraba because this emote is stored on humans and not buttcrabs. at least in theory.

I'll have a look into advert farting

**Comment by TheTyrant25** [comment]
> oh and before i forget, this wont cause other emotes to go on cooldown right? cause if im remembering right, FLIPPING cause all other emotes to not be used until its cooldown is over ((again, last i checked at least) i might be confusing uhhhh when critters emote but... just to be safe). and personally i find that rather annoying

it shouldnt do, that was a bug that got patched i think. at least i think so...

**Comment by deathrobotpunch** [comment]
if you are going to merge this, at least add a trait called prolific fatter that lowers the cooldown for farting that costs -2 trait points

**Comment by TheTyrant25** [comment]
> if you are going to merge this, at least add a trait called prolific farter that lowers the cooldown for farting that costs -2 trait points

~~forgot to say please~~

fart disbelief and prolific farter to be atomised out and not put here

**Comment by dwasint** [comment]
We are approaching dark times my friends.

**Comment by Sord213** [comment]
Oh. Got reopened then testmerged without discussion. very cool

**Comment by Xkeeper0** [comment]
Farts are something extremely classic to Goon and I don't like this. If it's really a problem I'd rather see the volume/effects of them reduced rather than the cooldown increased.

**Comment by RGBDeadSilent** [comment]
Riot! Tear it down! Tear it all down! Only a tyrant would remove our farts! Remove this PR and remove the tyranny over a clowns greatest weapon!

**Comment by TheColorCyan** [comment]
We clearly don't need to merge a PR that has 12 thumb downs on it with people in discord and forums also saying they dont want it...  The reopening and testmerging is pretty funny to me tho considering it had mostly negative feedback

**Comment by UrsulaMejor** [comment]
1. Farts are still in the game
2. This is, effectively, a buff to their effects, as testing has shown
3. Carpal tunnel required for engineering bad
4. Sound spam bad

**Comment by Xkeeper0** [comment]
It's still not a good change. If something has to change because of "carpal tunnel" _(really???)_ then the newer mechanic that requires it should be fixed and not something that's literally been in the game for 11 years, imvho.

If people don't like the sounds then that's why there's an emote volume adjustment.

**Comment by UrsulaMejor** [comment]
The volume is not the problem, it is the frequency. That's why this PR reduces the frequency, and not the volume.

**Comment by Xkeeper0** [comment]
Sooooo are we going to impose a five-second cooldown on every noise-making emote, or just this one that you don't like? Because so far every player feedback I've seen has been really, really negative about this change.

**Comment by UrsulaMejor** [comment]
Burps would be another one to look at

**Comment by UrsulaMejor** [comment]
The point of testmerges is to see if things stay negative after the initial "change bad" wave. If in a week, people realize this isn't a big deal and they like the buffs to the effects, we can keep it. If not, it can be un-test-merged. That is the point of test merges

**Comment by RGBDeadSilent** [comment]
> Burps would be another one to look at

No don't you dare!  Thats my emote! I can't live without spamming it!

**Comment by Xkeeper0** [comment]
You need to actually re-evaluate what the community thinks and not just go "there was no further negative feedback", because one does not mean the other. But _this change is so bad that it should not even be test merged_. That is my stance and I'm sticking to it -- this is a bad PR that is bad for the game.

**Comment by Xkeeper0** [comment]
god damnit I didn't intend to close this fucking github

**Comment by UrsulaMejor** [comment]
Most feedback in the discord appears positive to neutral, most feedback in game seems ambivalent to humorously joking without actually caring.

**Comment by UrsulaMejor** [comment]
Good PR that's good for the game 🤷‍♀️

**Comment by TheTyrant25** [comment]
there is a bit of a kneejerk reaction to anything touching the beloved farts. But there's a reason stuff like flipping has 5 second cooldown. Stuff that has physical effects/sound/visuals should have longer cooldowns imo to prevent spam. That is my view, and why I made it.

Most of the reasons i'm seeing for keeping it are just that; for keeping it the way it has been because that's what people are used to. If you flipped it around, that it was 5 seconds the whole time and someone suggested reducing it to 1, that would never ever get merged.

Anyway I guess we'll see.

**Comment by TheColorCyan** [comment]
> there is a bit of a kneejerk reaction to anything touching the beloved farts. But there's a reason stuff like flipping has 5 second cooldown. Stuff that has physical effects/sound/visuals should have longer cooldowns imo to prevent spam. That is my view, and why I made it.
> 
> Most of the reasons i'm seeing for keeping it are just that; for keeping it the way it has been because that's what people are used to. If you flipped it around, that it was 5 seconds the whole time and someone suggested reducing it to 1, that would never ever get merged.
> 
> Anyway I guess we'll see.

Stuff like flipping has cooldown first of all because of suplex mechanic and only then visual. Even for dancing there is a way to reduce the cooldown (dancing shoes).  Other visual stuff doesn't have long cooldowns but has short one just so it wouldn't be WAY to spammy. There is already way to prevent spam and longer cooldowns for combat related things like flipping. I feel like there is no need to increase it.
And yes... its kinda a thing to goon to have fart stuff and if you changed the way it was in the past of course people would have different opinions. I feel like its a silly thing to bring up.

**Comment by TheTyrant25** [comment]
farts are not the spirit of goonstation. we have many many other things

**Comment by MRLinew** [comment]
the farting is funny though its the soul of the game

**Comment by RSG250** [comment]
having actually played with it in testmerge now, i can say i REALLY dont like this now.

**Comment by TheTyrant25** [comment]
anyway this is your reminder that there is a forum post for this at https://forum.ss13.co/showthread.php?tid=21639

**Comment by Sord213** [comment]
> farts are not the spirit of goonstation. we have many many other things

damn, you heard it here first. the 2d spaceman farty game isn't known for the farty. I'm sticking with my opinion that this is a genuinely bad PR.

**Comment by Xkeeper0** [comment]
Flips have a larger cooldown because there are several mechanics that are tied to flips that break horribly if they don't have a long cooldown, specifically flipping over tables, suplexes, and matrix dodges. I personally think the flip cooldown is way too long, but it is what it is. (I'm also very much not a fan of whatever change made it so you couldn't flip in space or in the ocean!)

**Comment by flappybatpal** [comment]
I think this is a minor change that will have minimal real impact beyond reducing spam.
We still have a dedicated fart button people!

**Comment by Sbmhawk** [comment]
Having this be tied to the same cooldown as flips/screams just doesn't feel right. I genuinely don't think this is a good idea/PR

**Comment by flappybatpal** [comment]
Test merging is not pushing through. Test merging is temporary testing. A PR can still be decided to not be appropriate after a test merge.

**Comment by Sbmhawk** [comment]
I didnt mean pushed through literally, I was considering editing it to forced through.

**Comment by OmegaShoots** [comment]
add my -1 feedback to this
no i will not use the forum, no i will not use the discord

**Comment by braids** [comment]
> farts are not the spirit of goonstation. we have many many other things

this is an amazingly bad change

**Comment by zjdtmkhzt** [comment]
I love this change ^_^
best PR of the 21st century so far

**Comment by DabirSA** [comment]
What the hell is wrong with you? Undo this. Jesus Christ.

**Comment by TheTyrant25** [comment]
`If you are auto-staled anyway, ask developers if your PR will be merged.`

right. will this ever be merged.

**Comment by hobartbartholomewpepperidgejr** [comment]
after 4 months of seeing you necropost this one single pr i think its safe to say it's not getting merged

**Comment by UrsulaMejor** [comment]
Some other things might be done regarding the distance fart sounds travel, etc., but a cooldown change is unlikely.

**Pull request closed without merging**